vertx-pg-client/src/main/java/io/vertx/pgclient/impl/PgPoolImpl.java

```diff
@@ -40,11 +40,12 @@
  */
 public class PgPoolImpl extends PoolBase<PgPoolImpl> implements PgPool {
 
-  public static PgPoolImpl create(ContextInternal context, boolean closeVertx, PgConnectOptions connectOptions, PoolOptions poolOptions) {
+  public static PgPoolImpl create(ContextInternal context, boolean closeVertx, boolean pipelined, PgConnectOptions connectOptions, PoolOptions poolOptions) {
     QueryTracer tracer = context.tracer() == null ? null : new QueryTracer(context.tracer(), connectOptions);
     VertxMetrics vertxMetrics = context.owner().metricsSPI();
     ClientMetrics metrics = vertxMetrics != null ? vertxMetrics.createClientMetrics(connectOptions.getSocketAddress(), "sql", connectOptions.getMetricsName()) : null;
-    PgPoolImpl pool = new PgPoolImpl(context, new PgConnectionFactory(ConnectionFactory.asEventLoopContext(context), connectOptions), tracer, metrics, poolOptions);
+    int pipeliningLimit = pipelined ? connectOptions.getPipeliningLimit() : 1;
```

Comment:
If I am understanding it right, for a pool created using `Pool.pool` will behave like this after changes:

For Pool#getConnection operations, the query commands will be executed with pipeliningLimit=1(pipeliningLimit breaking change)
For Pool#query operations, the query commands will be scheduled in different pooled connections but be ing executed with pipeliningLimit=1(this is actually not a breaking change because currently our pool works in a way like this)
Please correct me if I'm wrong.

Reply:
yes, the goal is that connections are used exclusively and pipelining will only occur with the `client()` method